### PR TITLE
common: Fix ObjBencher clean_up when no prefix specified

### PR DESCRIPTION
**Pull request by dzafman**
Fix rados cleanup when there is no benchmark_last_metadata available
Support --all option with rados cleanup
Prevent rados cleanup from being used to remove non rados bench objects

Fixes: http://tracker.ceph.com/issues/15338

### CODE REVIEW

**Comment by dzafman** [comment]
The testing is part of this and runs during make check.

File: src/common/obj_bencher.cc [hunk 1]
```diff
@@ -1084,31 +1079,67 @@ int ObjBencher::rand_read_bench(int seconds_to_run, int num_objects, int concurr
...
+    r = fetch_bench_metadata(run_name_meta, &op_size, &object_size, &num_objects, &prevPid);
+    if (r < 0) {
       return r;
     }
-  }
 
-  r = clean_up(num_objects, prevPid, concurrentios);
-  if (r != 0) return r;
+    r = clean_up(num_objects, prevPid, concurrentios);
+    if (r != 0) return r;
```
Comment (athanatos) [inline src/common/obj_bencher.cc]: Whitespace thing?

File: src/common/obj_bencher.cc [hunk 2]
```diff
@@ -1084,31 +1079,67 @@ int ObjBencher::rand_read_bench(int seconds_to_run, int num_objects, int concurr
...
 
-  r = sync_remove(run_name_meta);
-  if (r != 0) return r;
+    r = sync_remove(run_name_meta);
+    if (r != 0) return r;
+  }
```
Comment (athanatos) [inline src/common/obj_bencher.cc]: Same here?

**Comment by athanatos** [comment]
lgtm

File: src/common/obj_bencher.cc (hunk 2 above)
Comment (dzafman) [inline src/common/obj_bencher.cc]: @athanatos Yes, this and the previous case are now inside a while loop, so are indented 1 additional level.

**Pull request merged**